modules/fraction/test/test_fraction.cpp

```diff
@@ -0,0 +1,335 @@
+    EXPECT_EQ(d, 15);
+}
+
+TEST_F(FractionTest, Denominator_Cant_Be_Negative) {
+    // Arrange
+    Fraction f1(1, -1);
+    Fraction f2(1);
+
+    // Act
+    f2.setDenominator(-1);
+
+    // Assert
+    EXPECT_GT(f1.getDenominator(), 0);
+    EXPECT_GT(f2.getDenominator(), 0);
+}
+
+TEST_F(FractionTest, Correct_Fraction_Reduction) {
+    // Arrange
+    Fraction f(5, 10);
+
+    // Act
+    f.fractionReduction();
+
+    // Assert
+    EXPECT_EQ(f.getNominator(), 1);
+    EXPECT_EQ(f.getDenominator(), 2);
+}
+
+TEST_F(FractionTest, Addition_Fractions_Test) {
+    // Arrange
+    Fraction f1(1, 15);
+    Fraction f2(3, 5);
+
+    // Act
+    Fraction f3 = f1 + f2;
+
+    // Assert
+    EXPECT_EQ(f3.getNominator(), 2);
+    EXPECT_EQ(f3.getDenominator(), 3);
+}
+
+TEST_F(FractionTest, Subtraction_Fractions_Test) {
+    // Arrange
+    Fraction f1(1, 15);
+    Fraction f2(3, 5);
+
+    // Act
+    Fraction f3 = f1 - f2;
+
+    // Assert
+    EXPECT_EQ(f3.getNominator(), -8);
+    EXPECT_EQ(f3.getDenominator(), 15);
+}
+
+TEST_F(FractionTest, Multiplication_Fractions_Test) {
+    // Arrange
+    Fraction f1(1, 15);
+    Fraction f2(3, 5);
+
+    // Act
+    Fraction f3 = f1 * f2;
+
+    // Assert
+    EXPECT_EQ(f3.getNominator(), 1);
+    EXPECT_EQ(f3.getDenominator(), 25);
+}
+
+TEST_F(FractionTest, Division_Fractions_Test) {
+    // Arrange
+    Fraction f1(1, 15);
+    Fraction f2(3, 5);
+
+    // Act
+    Fraction f3 = f1 / f2;
+
+    // Assert
+    EXPECT_EQ(f3.getNominator(), 1);
+    EXPECT_EQ(f3.getDenominator(), 9);
+}
+
+TEST_F(FractionTest, Assignment_Test) {
+    // Arrange
+    Fraction f1(1, 1);
+
+    // Act
+    Fraction f2 = f1;
+
+    // Assert
+    EXPECT_EQ(f1, f2);
+}
+
+TEST_F(FractionTest, Addition_Fractions_With_Int_Test) {
+    // Arrange
+    Fraction f1(1, 15);
+    int n = 5;
+
+    // Act
+    Fraction f2 = f1 + n;
+    Fraction f3 = n + f1;
+
+    // Assert
+    EXPECT_EQ(f2, Fraction(76, 15));
+    EXPECT_EQ(f3, Fraction(76, 15));
+}
+
+TEST_F(FractionTest, Subtraction_Fractions_With_Int_Test) {
+    // Arrange
+    Fraction f1(1, 15);
+    int n = 5;
+
+    // Act
+    Fraction f2 = f1 - n;
+    Fraction f3 = n - f1;
+
+    // Assert
+    EXPECT_EQ(f2, Fraction(-74, 15));
+    EXPECT_EQ(f3, Fraction(74, 15));
+}
+
+TEST_F(FractionTest, Multiplication_Fractions_With_Int_Test) {
+    // Arrange
+    Fraction f1(1, 15);
+    int n = 5;
+
+    // Act
+    Fraction f2 = f1 * n;
+    Fraction f3 = n * f1;
+
+    // Assert
+    EXPECT_EQ(f2, Fraction(1, 3));
+    EXPECT_EQ(f3, Fraction(1, 3));
+}
+
+TEST_F(FractionTest, Division_Fractions_With_Int_Test) {
+    // Arrange
+    Fraction f1(1, 15);
+    int n = 5;
+
+    // Act
+    Fraction f2 = f1 / n;
+    Fraction f3 = n / f1;
+
+    // Assert
+    EXPECT_EQ(f2, Fraction(1, 75));
+    EXPECT_EQ(f3, Fraction(75, 1));
+}
+
+TEST_F(FractionTest, Fraction_To_Double_Test) {
+    // Arrange
+    Fraction f1(1, 15);
+
+    // Act
+    double d = static_cast<double>(1) / 15;
```

Comment:
вещественные константы по умолчанию double
```suggestion
    double d = 1.0 / 15;
```